pkg/controller/podgc/gc_controller_test.go

```diff
@@ -657,6 +662,181 @@ func TestGCTerminating(t *testing.T) {
 	testDeletingPodsMetrics(t, 7)
 }
 
+func TestGCDeleteAllPodsBoundToNode(t *testing.T) {
+	testNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: "testNode"},
+		Status: v1.NodeStatus{
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceCPU):    resource.MustParse("10"),
+				v1.ResourceName(v1.ResourceMemory): resource.MustParse("10G"),
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                          string
+		node                          *v1.Node
+		pod                           *v1.Pod
+		expectedPatchedPod            *v1.Pod
+		expectedDeleteAction          *clienttesting.DeleteActionImpl
+		enablePodDisruptionConditions bool
+	}{
+		{
+			name: "orphaned pod should be deleted",
```

Comment:
```suggestion
			name: "orphaned pod should have DisruptionTarget condition added before deletion",
```
I think this it the main intention of the test case. The deletion itself is already covered in existing tests.

Reply:
Changed

Reply:
Feel free to resolve